src/main/scala/is/hail/io/LoadMatrix.scala

```diff
@@ -0,0 +1,231 @@
+package is.hail.io
+
+import is.hail.HailContext
+import is.hail.annotations._
+import is.hail.expr._
+import is.hail.sparkextras.OrderedRDD2
+import is.hail.utils._
+import is.hail.variant._
+import org.apache.hadoop.conf.Configuration
+import org.apache.spark.rdd.RDD
+
+import scala.language.implicitConversions
+import scala.io.Source
+
+object LoadMatrix {
+
+  def warnDuplicates(ids: Array[String]) {
+    val duplicates = ids.counter().filter(_._2 > 1)
+    if (duplicates.nonEmpty) {
+      warn(s"Found ${duplicates.size} duplicate ${plural(duplicates.size, "sample ID")}:\n  @1",
+        duplicates.toArray.sortBy(-_._2).map { case (id, count) => s"""($count) "$id"""" }.truncatable("\n  "))
+    }
+  }
+
+  // this assumes that col IDs are in last line of header.
+  /// FIXME: Is the toString.split call too slow?
```

Comment:
Seems fine here.  I'm mostly worried about per-cell performance and, to a lesser extent, per row (or per column) performance.